crates/cxx-qt-gen/src/writer/rust/mod.rs

```diff
@@ -76,33 +72,70 @@ fn cxx_common_blocks(
 
 /// For a given GeneratedRustBlocks write this into a Rust TokenStream
 pub fn write_rust(generated: &GeneratedRustBlocks) -> TokenStream {
-    // Retrieve the struct idents
-    let cpp_struct_ident = &generated.cpp_struct_ident;
-    let rust_struct_ident = &generated.rust_struct_ident;
-    let cxx_qt_thread_ident = &generated.cxx_qt_thread_ident;
-
     // Build the module idents
     let cxx_mod_ident = &generated.cxx_mod.ident;
     let cxx_qt_mod_ident = format_ident!("cxx_qt_{}", cxx_mod_ident);
 
     // Retrieve the module contents and namespace
     let mut cxx_mod = generated.cxx_mod.clone();
-    let cxx_qt_mod_contents = &generated.cxx_qt_mod_contents;
+    let mut cxx_qt_mod_contents = vec![];
     let namespace = &generated.namespace;
-    let namespace_internals = &generated.namespace_internals;
-
-    // Inject the common blocks into the bridge which we need
-    let cxx_mod_items = &mut cxx_mod.content.as_mut().expect("").1;
-    for block in cxx_common_blocks(
-        cpp_struct_ident,
-        rust_struct_ident,
-        cxx_qt_thread_ident,
-        namespace_internals,
-    ) {
-        cxx_mod_items.push(syn::parse2(block).expect("Could not build CXX common block"));
-    }
 
-    let create_rs_ident = mangle("create_rs", rust_struct_ident);
+    // Add comment includes for all objects
+    cxx_mod.content.as_mut().expect("").1.push(
+        syn::parse2(quote! {
+            unsafe extern "C++" {
+                include ! (< QtCore / QObject >);
+                include!("cxx-qt-lib/include/convert.h");
+                include!("cxx-qt-lib/include/cxxqt_thread.h");
+            }
+        })
+        .expect("Could not build CXX common block"),
+    );
+
+    for qobject in &generated.qobjects {
+        // Retrieve the struct idents
+        let cpp_struct_ident = &qobject.cpp_struct_ident;
+        let rust_struct_ident = &qobject.rust_struct_ident;
+        let cxx_qt_thread_ident = &qobject.cxx_qt_thread_ident;
+
+        let namespace_internals = &qobject.namespace_internals;
+
+        // Inject the common blocks into the bridge which we need
+        let cxx_mod_items = &mut cxx_mod.content.as_mut().expect("").1;
+        cxx_mod_items.extend_from_slice(&qobject.cxx_mod_contents);
+        for block in cxx_common_blocks(
+            cpp_struct_ident,
+            rust_struct_ident,
+            cxx_qt_thread_ident,
+            namespace_internals,
+        ) {
```

Comment:
Instead of extracting all these struct properties to pass to this function, how about refactoring `cxx_common_blocks` to take the `qobject`?

Reply:
Refactored that whole area, i think it looks better now :-)